lib/Jsrt/Jsrt.cpp

```diff
@@ -2791,44 +2791,65 @@ CHAKRA_API JsConstructObject(_In_ JsValueRef function, _In_reads_(cargs) JsValue
     });
 }
 
-CHAKRA_API JsCreateFunction(_In_ JsNativeFunction nativeFunction, _In_opt_ void *callbackState, _Out_ JsValueRef *function)
+#ifndef _CHAKRACOREBUILD
+struct JsNativeFunctionInfo
```

Comment:
Having this duplicated between the header and here feels like it could cause issues in future (e.g. someone updates one spot but not the other). Since it seems that this struct is required in both core and full (since it is used internally there), is it worth moving the struct definition to chakra.h instead of charkacore.h?

Reply:
I took the conservative approach of not adding this struct or the declaration for `JsEnhancedNativeFunction` in ChakraCommon.h because I don't want them to be considered part of the Windows API surface and have to go through the API review board, etc.

Reply:
Makes sense. I do wish there was a better way than having essentially the same structure duplicated in 3 places.

Reply:
Could it be in a separate header for the time being that is just included in each place? Its also a bit hacky but the pattern is used elsewhere in ChakraCore.